### PR TITLE
Added EditorImportPlugin class ref docs

### PR DESCRIPTION
**Pull request by rminderhoud**

### CODE REVIEW

File: doc/classes/EditorImportPlugin.xml [hunk 1]
```diff
@@ -1,8 +1,53 @@
...
+
+		EditorImportPlugins work by associating with specific file extensions (see: [method get_recognized_extension] and godot resource types (see: [method get_resource_type]). They may optionally specify some import presets that affect the import process. EditorImportPlugins are responsible for creating the resources and saving them in the [code].import[/code] directory.
+
+		Example:
+		[codeblock]
+		tool
+		extends EditorImportPlugin
+
+		func get_importer_name():
+			return "my.special.plugin"
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: Last tab should be 4 spaces.

File: doc/classes/EditorImportPlugin.xml [hunk 2]
```diff
@@ -1,8 +1,53 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="EditorImportPlugin" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+	   Used by the editor to import resources from files.
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: Indentation should be tabs.

File: doc/classes/EditorImportPlugin.xml (hunk 1 above)
Comment (rminderhoud) [reply doc/classes/EditorImportPlugin.xml]: Do you want me to change XML indentation to tabs and codeblock indentation to spaces?

Comment (Noshyaar) [reply doc/classes/EditorImportPlugin.xml]: Yes. https://github.com/GDquest/godot-docs/blob/3a2a243f335d584d41d028e33559ce3252fdd706/community/contributing/updating_the_class_reference.rst

File: doc/classes/EditorImportPlugin.xml [hunk 3]
```diff
@@ -45,33 +93,37 @@
...
 			</return>
 			<description>
+				Get the godot resource type associated with this loader. e.g. "Mesh" or "Animation".
 			</description>
 		</method>
 		<method name="get_save_extension" qualifiers="virtual">
 			<return type="String">
 			</return>
 			<description>
+				Get the extension used to save this resource in the [code].import[/code] directory
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: Missing period at the end.

File: doc/classes/EditorImportPlugin.xml [hunk 4]
```diff
@@ -15,12 +60,14 @@
 			<argument index="0" name="preset" type="int">
 			</argument>
 			<description>
+				Get the options and default values for the preset at this index. Returns an Array of Dictionaries with the following keys: "name", "default_value", "property_hint" (optional), "hint_string" (optional), "usage" (optional)
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: Missing period at the end.

File: doc/classes/EditorImportPlugin.xml [hunk 5]
```diff
@@ -45,33 +93,37 @@
...
 		</method>
 		<method name="get_visible_name" qualifiers="virtual">
 			<return type="String">
 			</return>
 			<description>
+				Get the name to display in the import window.
 			</description>
-		</method>
-		<method name="import" qualifiers="virtual">
+		</method> <method name="import" qualifiers="virtual">
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: This looks weird. It shouldn't be in the same line?

**Comment by rminderhoud** [comment]
All changes made as requested, thanks @Noshyaar

**Comment by n-pigeon** [comment]
@rminderhoud Nice! :D

File: doc/classes/EditorImportPlugin.xml [hunk 6]
```diff
@@ -1,8 +1,53 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="EditorImportPlugin" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Used by the editor to import resources from files.
 	</brief_description>
 	<description>
+		EditorImportPlugins provide a way to extend the editor's resource import functionality. They can be used to import resources from custom files or to provide alternatives to the editor's existing importers. EditorImportPlugins must be registered by an [EditorPlugin] by using [method EditorPlugin.add_import_plugin].
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: Avoid passive voice
`They can be used` -> `Use them to`
Use the direct voie
`EditorImportPlugins must be registered by an [EditorPlugin] by using [method EditorPlugin.add_import_plugin]` -> `Register your EditorImportPlugin with [method EditorPlugin.add_import_plugin]`

File: doc/classes/EditorImportPlugin.xml [hunk 7]
```diff
@@ -1,8 +1,53 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="EditorImportPlugin" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Used by the editor to import resources from files.
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: The brief desc is very important. Please give an overview of the node in the brief description.
Idea (not sure exactly what the class does, just an example):
`Registers custom importers in the editor. Use the class to parse any file and store it as a new ressource type.`

File: doc/classes/EditorImportPlugin.xml [hunk 8]
```diff
@@ -1,8 +1,53 @@
...
 	</brief_description>
 	<description>
+		EditorImportPlugins provide a way to extend the editor's resource import functionality. They can be used to import resources from custom files or to provide alternatives to the editor's existing importers. EditorImportPlugins must be registered by an [EditorPlugin] by using [method EditorPlugin.add_import_plugin].
+
+		EditorImportPlugins work by associating with specific file extensions (see: [method get_recognized_extension] and godot resource types (see: [method get_resource_type]). They may optionally specify some import presets that affect the import process. EditorImportPlugins are responsible for creating the resources and saving them in the [code].import[/code] directory.
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: Avoid parens: use short sentences instead. Exceptional case where you may use passive voice, as we can consider that EditorImportPlugin is linked to other resources by its parent class, the system, the editor...
`An EditorImportPlugin is linked to specific file extensions and resource type. See: [method get_recognized_extension] and [method get_resource_type].`

File: doc/classes/EditorImportPlugin.xml [hunk 9]
```diff
@@ -1,8 +1,53 @@
...
+
+		Example:
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: Could you add a few words to explain what the example shows?

File: doc/classes/EditorImportPlugin.xml [hunk 10]
```diff
@@ -15,12 +60,14 @@
...
 			</argument>
 			<description>
+				Get the options and default values for the preset at this index. Returns an Array of Dictionaries with the following keys: "name", "default_value", "property_hint" (optional), "hint_string" (optional), "usage" (optional).
 			</description>
 		</method>
 		<method name="get_importer_name" qualifiers="virtual">
 			<return type="String">
 			</return>
 			<description>
+				Get the unique name of the importer.
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: No need to document get/set methods: document the <member> tags instead. get/setters will disappear in a future version of the docs (after we moved all existing descriptions to members)

File: doc/classes/EditorImportPlugin.xml [hunk 11]
```diff
@@ -45,30 +93,35 @@
...
 			</argument>
 			<description>
+				Get the name of the options preset at this index.
 			</description>
 		</method>
 		<method name="get_recognized_extensions" qualifiers="virtual">
 			<return type="Array">
 			</return>
 			<description>
+				Get the list of file extensions to associate with this loader (case insensitive). e.g. ["obj"].
```
Comment (NathanLovato) [inline doc/classes/EditorImportPlugin.xml]: For returned values, start with "returns". Reuse words that best describe the class over synonyms: loader -> importer or import plugin. Also avoid parens.
Returns the list of file extensions to associate with this import plugin. Extensions are not case sensitive. E.g. ["obj"].

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Great work, but needs some changes to get a merge. Please read the writing guidelines: https://github.com/GDquest/godot-3-guides/blob/master/contribution-guides/class-reference-styleguide.md

File: doc/classes/EditorImportPlugin.xml (hunk 10 above)
Comment (rminderhoud) [reply doc/classes/EditorImportPlugin.xml]: Can you clarify what you mean by documenting the tag?

File: doc/classes/EditorImportPlugin.xml (hunk 9 above)
Comment (rminderhoud) [reply doc/classes/EditorImportPlugin.xml]: Is `<description>` the right section for the example? I see `<tutorials>` and `<demos>` as well..

File: doc/classes/EditorImportPlugin.xml (hunk 10 above)
Comment (NathanLovato) [reply doc/classes/EditorImportPlugin.xml]: The member tags - turns out github treats \<member\> as an empty url so it hid it ^^

File: doc/classes/EditorImportPlugin.xml (hunk 9 above)
Comment (NathanLovato) [reply doc/classes/EditorImportPlugin.xml]: Yes. In tutorial and demo, we can put URLs to pages from the docs.

File: doc/classes/EditorImportPlugin.xml [hunk 12]
```diff
@@ -45,30 +94,35 @@
...
 			</description>
 		</method>
 		<method name="get_resource_type" qualifiers="virtual">
 			<return type="String">
 			</return>
 			<description>
+				Get the godot resource type associated with this loader. e.g. "Mesh" or "Animation".
```
Comment (Noshyaar) [inline doc/classes/EditorImportPlugin.xml]: >godot

`Godot` 😢

File: doc/classes/EditorImportPlugin.xml (hunk 10 above)
Comment (rminderhoud) [reply doc/classes/EditorImportPlugin.xml]: Actually as far as I can see there are no `<member>`s tag in this document...

File: doc/classes/EditorImportPlugin.xml (hunk 12 above)
Comment (rminderhoud) [reply doc/classes/EditorImportPlugin.xml]: Oh no...looks like my left pinkie finger was slacking :(

File: doc/classes/EditorImportPlugin.xml [hunk 13]
```diff
@@ -1,8 +1,54 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="EditorImportPlugin" inherits="Reference" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Registers a custom resource importer in the editor. Use the class to parse any file and import it as a new resource type.
 	</brief_description>
 	<description>
+		EditorImportPlugins provide a way to extend the editor's resource import functionality. Use them to import resources from custom files or to provide alternatives to the editor's existing importers. Register your [EditorPlugin] with [method EditorPlugin.add_import_plugin].
+
```
Comment (akien-mga) [inline doc/classes/EditorImportPlugin.xml]: Please remove the empty lines, as they would be parsed as extraneous paragraphs in the editor and the online docs. Each line of the XML is a paragraph (apart from within codeblocks which work more as preformatted text).

**Comment by akien-mga** [comment]
Well, good enough for now, need to clear the PR backlog anyway...

**Pull request merged**